src/EditorFeatures/TestUtilities/Threading/WpfTestRunner.cs

```diff
@@ -50,39 +50,67 @@ public sealed class WpfTestRunner : XunitTestRunner
         protected override Task<decimal> InvokeTestMethodAsync(ExceptionAggregator aggregator)
         {
             SharedData.ExecutingTest(TestMethod);
-            var sta = StaTaskScheduler.DefaultSta;
+
+            DispatcherSynchronizationContext synchronizationContext = null;
+            Dispatcher dispatcher = null;
+            var staThreadStartedEvent = new ManualResetEventSlim(initialState: false);
+            var staThread = new Thread((ThreadStart)(() =>
+            {
+                // All WPF Tests need a DispatcherSynchronizationContext and we dont want to block pending keyboard
+                // or mouse input from the user. So use background priority which is a single level below user input.
+                synchronizationContext = new DispatcherSynchronizationContext();
```

Comment:
Why aren't we restoring the old context anymore?

Reply:
This was part of the revert commit. The thread is terminating so it doesn't seem necessary to restore it.